variant_to_realignment.nf

```diff
@@ -0,0 +1,215 @@
+    # [seq]
+    sed 's/\\t/|/g; s/ /|/g; s/\\(.*\\)|/\\1 /' temp1.txt | tr ' ' '\\n' > variant_read1.out.fa
+    sed 's/\\t/|/g; s/ /|/g; s/\\(.*\\)|/\\1 /' temp2.txt | tr ' ' '\\n' > variant_read2.out.fa
+    '''
+}
+
+/*
+ * Align sequence with minimap2
+ */
+process alignWithMinimap {
+
+    // Memory required is 5 times the size of the fasta in Bytes or at least 1GB
+    memory Math.max(file(params.newgenome).size() * 5, 1073741824) + ' B'
+
+    input:
+        path "variant_read1.fa"
+        path "variant_read2.fa"
+        // indexing is done on the fly so get the genome directly
+        path 'genome.fa'
+
+    output:
+        path "reads_aligned.bam", emit: reads_aligned_bam
+
+    """
+    # Options used by the 'sr' preset but allowing secondary alignments
+    minimap2 -k21 -w11 --sr --frag=yes -A2 -B5 -O6,16 --end-bonus 20 -E2,1 -r50 -p.5 -z 800,200\
+             -f1000,5000 -n2 -m20 -s40 -g200 -2K50m --heap-sort=yes --secondary=yes -N 2 \
+             -a genome.fa variant_read1.fa variant_read2.fa | samtools view -bS - > reads_aligned.bam
+    """
+}
+
+/*
+ * Take the reads and process them to get the remapped variants
+ *
+ */
+process readsToRemappedVariants {
+
+    input:
+        path "reads_aligned.bam"
+        path "genome.fa"
+
+    output:
+        path "variants_remapped.vcf", emit: variants_remapped
+        path "variants_unmapped.vcf", emit: variants_unmapped
+
+    """
+    # Ensure that we will use the reads_to_remapped_variants.py from this repo
+    ${baseDir}/variant_remapping_tools/reads_to_remapped_variants.py -i reads_aligned.bam \
+        -o variants_remapped.vcf  --filter_align_with_secondary \
+        --newgenome genome.fa --out_failed_file variants_unmapped.vcf
+    """
+}
+
+workflow process_split_reads {
+    take:
+        source_vcf
+        old_genome_fa
+        old_genome_fa_fai
+        old_genome_chrom_sizes
+        new_genome_fa
+        new_genome_fa_fai
+
+    main:
+        ConvertVCFToBed(source_vcf)
+        flankingRegionBed(ConvertVCFToBed.out.variants_bed, old_genome_chrom_sizes)
+        flankingRegionFasta(
+            flankingRegionBed.out.flanking_r1_bed, flankingRegionBed.out.flanking_r2_bed,
+            old_genome_fa, old_genome_fa_fai
+        )
+        extractVariantInfoToFastaHeader(
+            flankingRegionBed.out.flanking_r1_bed, flankingRegionBed.out.flanking_r2_bed,
+            flankingRegionFasta.out.variants_read1, flankingRegionFasta.out.variants_read2
+        )
+        alignWithMinimap(
+            extractVariantInfoToFastaHeader.out.variant_read1_with_info,
+            extractVariantInfoToFastaHeader.out.variant_read2_with_info,
+            new_genome_fa
+        )
+        readsToRemappedVariants(alignWithMinimap.out.reads_aligned_bam, new_genome_fa)
+
+    emit:
+        variants_remapped = readsToRemappedVariants.out.variants_remapped
+        variants_unmapped = readsToRemappedVariants.out.variants_unmapped
+}
```

Comment:
And a final question, did you run the precision/recall calculation on this pipeline after the changes? I would be interested to see the metrics

Reply:
I did the current state is similar a to what I was getting with bowtie2:

Type | Filter | TRUTH.TOTAL | TRUTH.TP | TRUTH.FN | QUERY.TOTAL | QUERY.FP | QUERY.UNK | FP.gt | METRIC.Recall | METRIC.Precision
-- | -- | -- | -- | -- | -- | -- | -- | -- | -- | --
INDEL | ALL | 383937 | 325483 | 58454 | 325529 | 46 | 0 | 0 | 0.847751 | 0.999859
SNP | ALL | 3254376 | 2685067 | 569309 | 2685075 | 8 | 0 | 0 | 0.825064 | 0.999997

Reply:
Thanks! Agreed, seems similar